components/authz-service/storage/postgres/migration/sql/53_add_db_id_to_iam_statements.up.sql

```diff
@@ -0,0 +1,187 @@
+BEGIN;
+ALTER TABLE iam_policy_members
+    DROP CONSTRAINT iam_policy_members_policy_id_fkey;
+ALTER TABLE iam_policy_projects
+    DROP CONSTRAINT iam_policy_projects_policy_id_fkey;
+ALTER TABLE iam_policy_statements
+    DROP CONSTRAINT iam_policy_statements_policy_id_fkey;
+
+ALTER TABLE iam_policies
+    DROP CONSTRAINT iam_policies_pkey;
+ALTER TABLE iam_policies
+    ADD COLUMN db_id SERIAL PRIMARY KEY UNIQUE;
+ALTER TABLE iam_policies
+    ADD UNIQUE (id);
+
+ALTER TABLE iam_policy_members RENAME COLUMN policy_id TO policy_temp_id;
+ALTER TABLE iam_policy_members
+    ADD COLUMN policy_id INTEGER REFERENCES iam_policies (db_id) ON DELETE CASCADE DEFERRABLE;
+UPDATE
+    iam_policy_members ipm
+SET
+    policy_id = (
+        SELECT
+            db_id
+        FROM
+            iam_policies
+        WHERE
+            id = ipm.policy_temp_id);
+ALTER TABLE iam_policy_members
+    DROP COLUMN policy_temp_id;
+ALTER TABLE iam_policy_members
+    ADD UNIQUE (member_id, policy_id);
+
+ALTER TABLE iam_policy_projects RENAME COLUMN policy_id TO policy_temp_id;
+ALTER TABLE iam_policy_projects
+    ADD COLUMN policy_id INTEGER REFERENCES iam_policies (db_id) ON DELETE CASCADE;
+UPDATE
+    iam_policy_projects ipp
```

Comment:
side-note: it'd be great if we could add a db diagram to one of our readmes once this db_id cleanup is done. [this site](https://dbdiagram.io/home) looked promising - i think we might be able to import our postgresql

Reply:
ooooo